### PR TITLE
sql/mysql: support marshaling generated columns

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/mysql/driver.go [hunk 1]
```diff
@@ -282,7 +282,7 @@ const (
 	defaultGen    = "default_generated"
 	autoIncrement = "auto_increment"
 
-	genVirtual    = "virtual"
-	genStored     = "stored"
-	genPersistent = "persistent"
+	genVirtual    = "VIRTUAL"
+	genStored     = "STORED"
+	genPersistent = "PERSISTENT"
```
Comment (masseelch) [inline sql/mysql/driver.go:287]: Why `gen` prefix?

File: sql/mysql/sqlspec.go [hunk 2]
```diff
@@ -256,9 +256,28 @@ func columnSpec(c *schema.Column, t *schema.Table) (*sqlspec.Column, error) {
...
+	}
+	spec.Extra.Children = append(spec.Extra.Children, &schemaspec.Resource{
+		Type: "as",
+		Attrs: []*schemaspec.Attr{
+			specutil.VarAttr("type", t),
+			specutil.StrAttr("expr", x.Expr),
+		},
+	})
+}
+
```
Comment (masseelch) [inline sql/mysql/sqlspec.go:280]: Shouldn't the internal representation be the same, regardless of what was defined in HCL? And I think being explicit is cleaner, so if we marshal it wen can always do the `Resource` way.

**Review by masseelch — APPROVED** [review]

File: sql/mysql/sqlspec.go (hunk 2 above)
Comment (a8m) [reply sql/mysql/sqlspec.go:280]: Talked to it @rotemtam. I'll move it to the `specutil` after I'll need it in PostgreSQL.

Comment (a8m) [reply sql/mysql/sqlspec.go:280]: >  And I think being explicit is cleaner, so if we marshal it wen can always do the Resource way.

Agree.

File: sql/mysql/driver.go (hunk 1 above)
Comment (a8m) [reply sql/mysql/driver.go]: No special reason. I love grouping/namespacing constants that are related to each other. I can remove the prefix.

**Pull request merged**